server/channels/app/analytics.go

```diff
@@ -34,287 +34,273 @@ func (a *App) getAnalytics(rctx request.CTX, name string, teamID string, forSupp
-		g.Go(func() error {
-			var err error
-			if privateChannelsCount, err = a.Srv().Store().Channel().AnalyticsTypeCount(teamID, model.ChannelTypePrivate); err != nil {
-				return model.NewAppError("GetAnalytics", "app.channel.analytics_type_count.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-			}
-			return nil
-		})
-
-		var usersCount int64
-		var inactiveUsersCount int64
-		if teamID == "" {
-			g.Go(func() error {
-				var err error
-				if inactiveUsersCount, err = a.Srv().Store().User().AnalyticsGetInactiveUsersCount(); err != nil {
-					return model.NewAppError("GetAnalytics", "app.user.analytics_get_inactive_users_count.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-				}
-				return nil
-			})
-		} else {
-			g.Go(func() error {
-				var err error
-				if usersCount, err = a.Srv().Store().User().Count(model.UserCountOptions{TeamId: teamID}); err != nil {
-					return model.NewAppError("GetAnalytics", "app.user.get_total_users_count.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-				}
-				return nil
-			})
-		}
+	switch name {
+	case "standard":
+		return a.getStandardAnalytics(rctx, teamID, systemUserCount)
+	case "bot_post_counts_day":
+		return a.getBotPostCountsAnalytics(rctx, teamID)
+	case "post_counts_day":
+		return a.getPostCountsAnalytics(rctx, teamID)
+	case "user_counts_with_posts_day":
+		return a.getUserCountsWithPostsAnalytics(rctx, teamID, skipIntensiveQueries)
+	case "extra_counts":
+		return a.getExtraCountsAnalytics(rctx, teamID)
+	default:
+		return nil, nil
+	}
+}
 
-		var postsCount int64
-		if !skipIntensiveQueries {
-			g.Go(func() error {
-				var err error
-				if postsCount, err = a.Srv().Store().Post().AnalyticsPostCount(&model.PostCountOptions{TeamId: teamID}); err != nil {
-					return model.NewAppError("GetAnalytics", "app.post.analytics_posts_count.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
-				}
-				return nil
-			})
+func (a *App) getStandardAnalytics(rctx request.CTX, teamID string, systemUserCount int64) (model.AnalyticsRows, *model.AppError) {
+	var rows model.AnalyticsRows = make([]*model.AnalyticsRow, 11)
+	rows[0] = &model.AnalyticsRow{Name: "channel_open_count", Value: 0}
+	rows[1] = &model.AnalyticsRow{Name: "channel_private_count", Value: 0}
+	rows[2] = &model.AnalyticsRow{Name: "post_count", Value: 0}
+	rows[3] = &model.AnalyticsRow{Name: "unique_user_count", Value: 0}
+	rows[4] = &model.AnalyticsRow{Name: "team_count", Value: 0}
+	rows[5] = &model.AnalyticsRow{Name: "total_websocket_connections", Value: 0}
+	rows[6] = &model.AnalyticsRow{Name: "total_master_db_connections", Value: 0}
+	rows[7] = &model.AnalyticsRow{Name: "total_read_db_connections", Value: 0}
+	rows[8] = &model.AnalyticsRow{Name: "daily_active_users", Value: 0}
+	rows[9] = &model.AnalyticsRow{Name: "monthly_active_users", Value: 0}
+	rows[10] = &model.AnalyticsRow{Name: "inactive_user_count", Value: 0}
+
+	var g errgroup.Group
+	g.SetLimit(2)
```

Comment:
Should we add a disclaimer here stating that unlimited concurrency actually hurts performance?

Reply:
Don't think it needs to be explicitly stated. Because it's already mentioned here: https://developers.mattermost.com/contribute/more-info/server/style-guide/#performance-sensitive-areas